src/redux-middleware/pushQueue.ts

```diff
@@ -139,13 +133,54 @@ const flushPushQueue = (): Thunk<Promise<void>> => async (dispatch, getState) =>
 
   if (pushQueue.length === 0) return Promise.resolve()
 
-  dispatch(flushDeletes())
-  dispatch(flushEdits())
-  dispatch(flushMoves())
+  /**
+   * Returns updated thoughtIndexUpdates using the latest state after pulling pending lexemes.
+   */
+  const getSyncedThoughtsUpdate = (batch: PushBatch) => Object.keys(batch.thoughtIndexUpdates).reduce<Index<Lexeme | null>>((acc, thoughtId) => {
```

Comment:
Instead of taking the whole `batch` you can just take `pendingLexemes` and `thoughtIndexUpdates`, which is a narrower dependency.

Reply:
Sure.